privatevcs/validation/blocklist/rule.go

```diff
@@ -0,0 +1,62 @@
+package blocklist
+
+import (
+	"net/http"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/spacelift-io/vcs-agent/privatevcs/validation"
+)
+
+// Rule is a single rule in a blocklist.
+type Rule struct {
+	// The name of the rule.
+	Name string `json:"name"`
+
+	// The HTTP method to match.
+	Method string `json:"method"`
+
+	// The regular expression to match the path against.
+	Path string `json:"path"`
+
+	// The regular expression to match the vendor against.
+	Vendor string `json:"vendor"`
```

Comment:
I think adding vendor at the rule level is maybe overcomplicating things to begin with - at the moment we need to specify the vendor globally for a single instance of the agent, so it wouldn't make sense to set the vendor to GitHub and allow a mixture of GitHub and GitLab rules for example.

Reply:
Fair.